Content.Server/_DV/Curation/Systems/CwoinkSystem.cs

```diff
@@ -0,0 +1,1021 @@
+
+                        RaiseNetworkEvent(new SharedCwoinkSystem.CwoinkTextMessage(message.UserId,
+                                senderId,
+                                overrideMsgText,
+                                playSound: playSound),
+                            session.Channel);
+                    }
+                    else
+                        RaiseNetworkEvent(msg, session.Channel);
+                }
+            }
+
+            var sendsWebhook = _webhookUrl != string.Empty;
+            if (sendsWebhook && sendWebhook)
+            {
+                if (!_messageQueues.ContainsKey(msg.UserId))
+                    _messageQueues[msg.UserId] = new Queue<DiscordRelayedData>();
+
+                var str = message.Text;
+                var unameLength = senderName.Length;
+
+                if (unameLength + str.Length + _maxAdditionalChars > DescriptionMax)
+                {
+                    str = str[..(DescriptionMax - _maxAdditionalChars - unameLength)];
+                }
+
+                var nonAfkAdmins = GetNonAfkAdmins();
+                var messageParams = new CHelpMessageParams(
+                    senderName,
+                    str,
+                    senderId != message.UserId,
+                    _gameTicker.RoundDuration().ToString("hh\\:mm\\:ss"),
+                    _gameTicker.RunLevel,
+                    playedSound: playSound,
+                    isDiscord: fromWebhook, // DeltaV
+                    curatorOnly: message.AdminOnly,
+                    noReceivers: nonAfkAdmins.Count == 0
+                );
+                _messageQueues[msg.UserId].Enqueue(GenerateAHelpMessage(messageParams, _discordReplyPrefix));
+            }
+
+            if (admins.Count != 0 || sendsWebhook)
+                return;
+
+            // No admin online, let the player know
+            if (senderChannel != null)
+            {
+                var systemText = Loc.GetString("cwoink-system-starmute-message-no-other-users");
+                var starMuteMsg = new SharedCwoinkSystem.CwoinkTextMessage(message.UserId, SystemUserId, systemText);
+                RaiseNetworkEvent(starMuteMsg, senderChannel);
+            }
+        }
+        // End Frontier:
+
+        private IList<INetChannel> GetNonAfkAdmins()
+        {
+            return _adminManager.ActiveAdmins
+                .Where(p => (_adminManager.GetAdminData(p)?.HasFlag(AdminFlags.CuratorHelp) ?? false) &&
+                            !_afkManager.IsAfk(p))
+                .Select(p => p.Channel)
+                .ToList();
+        }
+
+        private IList<INetChannel> GetTargetAdmins()
+        {
+            return _adminManager.ActiveAdmins
+                .Where(p => _adminManager.GetAdminData(p)?.HasFlag(AdminFlags.CuratorHelp) ?? false)
+                .Select(p => p.Channel)
+                .ToList();
```

Comment:
do not the linq, store a list on the system and clear then add relevant sessions